core/src/main/java/org/apache/iceberg/Partitioning.java

```diff
@@ -195,41 +198,68 @@ public Void alwaysNull(int fieldId, String sourceName, int sourceId) {
   }
 
   /**
-   * Builds a common partition type for all specs in a table.
+   * Builds an intersection of all partition types in a table.
    *
-   * <p>Whenever a table has multiple specs, the partition type is a struct containing all columns
-   * that have ever been a part of any spec in the table.
+   * <p>Whenever a table has multiple specs, the common partition type is a struct containing only
+   * fields that are present in every spec of the table. In other words, the struct fields represent
+   * an intersection of all partition types.
    *
-   * @param table a table with one or many specs
+   * @param specs one or many specs
    * @return the constructed common partition type
```

Comment:
If I remember correctly, void transforms are only used in v1 tables to replace existing fields since we can't drop them. We want to use this as a clustering key in storage-partitioned joins, meaning void transforms can't be returned.

Reply:
I think that's fine, just needs to be doc'd

Reply:
I've changed the naming and logic, documented the behavior.